indico/MaKaC/PDFinterface/conference.py

```diff
@@ -643,715 +632,742 @@ def _processTOCPage(self):
-                for c in contrib.getSpeakerList():
-                    if self._showSpeakerAffiliation and c.getAffiliation().strip() != "":
-                        spks.append("%s (%s)"%(escape(c.getFullName()), escape(c.getAffiliation())))
+            # contribution
+            elif self._ttPDFFormat.showContribsAtConfLevel() and entry.type == TimetableEntryType.CONTRIBUTION:
+                contrib = entry.object
+                if not contrib.can_access(self._aw):
+                    continue
+
+                room = ''
+                if contrib.location_data['room_name']:
+                    room = ' - {}'.format(escape(contrib.location_data['room_name']))
+
+                speakers = []
+                for c in contrib.speakers:
+                    person = c.person
+                    if self._showSpeakerAffiliation and person.affiliation.strip():
+                        speakers.append('{} ({})'.format(escape(person.full_name), escape(person.affiliation)))
                     else:
-                        spks.append(escape(c.getFullName()))
-                spks="; ".join(spks)
-                if spks.strip()!="":
-                    spks=i18nformat("""<font name=\"Times-Bold\"><b>- _("Presenters"): %s</b></font>""")%spks
-                text="""<u>%s</u>%s (%s-%s)"""%(
-                        escape(caption),room,
-                        escape(contrib.getAdjustedStartDate(self._tz).strftime("%H:%M")),
-                        escape(contrib.getAdjustedEndDate(self._tz).strftime("%H:%M")))
-                p1=Paragraph(text,self._styles["session_title"])
+                        speakers.append(escape(person.affiliation))
+
+                speakers = '; '.join(speakers)
+                if speakers.strip():
+                    speakers = i18nformat('<font name="Times-Bold"><b>-_("Presenters"): {}</b></font>').format(speakers)
+
+                text = '<u>{}</u>{} ({}-{})'.format(escape(contrib.title), room,
+                                                    format_time(entry.start_dt, timezone=self._tz),
+                                                    format_time(entry.end_dt, timezone=self._tz))
+                p1 = Paragraph(text, self._styles["session_title"])
                 res.append(p1)
                 if self._ttPDFFormat.showTitleSessionTOC():
-                    self._indexedFlowable[p1]={"text":escape(contrib.getTitle()), "level":2}
-                p2=Paragraph(spks,self._styles["conveners"])
+                    self._indexedFlowable[p1] = {'text': escape(contrib.title), 'level': 2}
+
+                p2 = Paragraph(speakers, self._styles["conveners"])
                 res.append(p2)
-                if entry == entriesOnDay[-1]: # if it is the last one, we do the page break and remove the previous one.
-                    i=len(res)-1
-                    while i>=0:
+                if entry == entries[-1]:  # if it is the last one, we do the page break and remove the previous one.
```

Comment:
I noticed only now that we're still inside the same for loop. Yay for legacy code!